pinot-core/src/main/java/org/apache/pinot/core/transport/AsyncQueryResponse.java

```diff
@@ -145,6 +146,21 @@ public String getServerStats() {
     return stringBuilder.toString();
   }
 
+  @Override
+  public Map<String, Map<String, Integer>> getServerStatsMap() {
+    return _responseMap.entrySet().stream()
+        .collect(Collectors.toMap(
+            entry -> entry.getKey().getShortName(),
```

Comment:
The key here is the host right? is there a way to make it more explicit? Maybe instead of using Map<String, Map<String, Integer>> we can use Map<String, ServerStatsInfo> something and use POJO to easily extract the metric values.